proof/crefine/RISCV64/ADT_C.thy

```diff
@@ -1176,62 +1176,103 @@ lemma scRefills_unique:
   done
 
 lemma scSize_unique:
-  assumes "refill_buffer_relation ah ch gs" "refill_buffer_relation ah' ch gs"
-      and "ksPSpace s = ah \<and> ksPSpace s p = Some (KOSchedContext sc)"
-      and "ksPSpace s' = ah' \<and> ksPSpace s' p = Some (KOSchedContext sc')"
+  assumes "refill_buffer_relation (ksPSpace as) ch gs" "refill_buffer_relation (ksPSpace as') ch gs"
+      and "ksPSpace as p = Some (KOSchedContext sc)"
+      and "ksPSpace as' p = Some (KOSchedContext sc')"
   shows   "scSize sc = scSize sc'"
   apply (insert assms)
-  apply (clarsimp simp: valid_sched_context'_def)
-  apply (frule refill_buffer_relation_size_eq[where ah="ksPSpace s"])
-  apply (frule refill_buffer_relation_size_eq[where ah="ksPSpace s'"])
+  apply (frule refill_buffer_relation_size_eq[where ah="ksPSpace as"])
+  apply (frule refill_buffer_relation_size_eq[where ah="ksPSpace as'"])
   apply (clarsimp simp: Let_def)
   apply (drule_tac x=p in spec)+
   by fastforce
 
+lemma
+  assumes "pspace_aligned' as" "pspace_distinct' as" "valid_sched_context' asc as"
+  shows tcb_at'_scTCB:
+    "bound (scTCB asc) \<longrightarrow> tcb_at' (the (scTCB asc)) as"
+  and reply_at'_scReply:
+    "bound (scReply asc) \<longrightarrow> reply_at' (the (scReply asc)) as"
```

Comment:
`bound` is the wrong predicate here (there is nothing bound, it's just an option pointer). `is_Some` would work, but it looks to me like avoiding `the` would be even better.

Reply:
I've changed this to use the `!tcbPtr. scTCB asc = Some tcbPtr --> tcb_at' tcbPtr as` format in all those statements